igor-web/src/main/groovy/com/netflix/spinnaker/igor/build/BuildController.groovy

```diff
@@ -55,24 +55,17 @@ class BuildController {
     Map getJobStatus(@PathVariable String master, @PathVariable Integer buildNumber, HttpServletRequest request) {
         def job = (String) request.getAttribute(
             HandlerMapping.PATH_WITHIN_HANDLER_MAPPING_ATTRIBUTE).split('/').drop(5).join('/')
-        if (buildMasters.filteredMap(BuildServiceProvider.JENKINS).containsKey(master)) {
-            Map result = objectMapper.convertValue(buildMasters.map[master].getBuild(job, buildNumber), Map)
+        if (buildMasters.map.containsKey(master)) {
+            Map result = objectMapper.convertValue(buildMasters.map[master].getGenericBuild(job, buildNumber), Map)
             try {
-                Map scm = objectMapper.convertValue(buildMasters.map[master].getGitDetails(job, buildNumber), Map)
-                if (scm?.action?.lastBuiltRevision?.branch?.name) {
-                    result.scm = scm?.action.lastBuiltRevision
-                    result.scm = result.scm.branch.collect {
-                        it.branch = it.name.split('/').last()
-                        it
-                    }
+                def scm = objectMapper.convertValue(buildMasters.map[master].getGenericGitRevision(job, buildNumber), Map)
+                if (scm) {
+                    result.scm = scm
```

Comment:
Ah, so `scm = null` is no good in the response? No need to stress about this, we can leave it as is